src/Core/src/Response.php

```diff
@@ -87,18 +92,17 @@ public function __destruct()
             throw $this->resolveResult = new NetworkException('Could not contact remote server.', 0, $e);
         }
 
-        if ($throw) {
-            if (500 <= $statusCode) {
-                throw $this->resolveResult = new ServerException($this->response);
-            }
-
-            if (400 <= $statusCode) {
-                throw $this->resolveResult = new ClientException($this->response);
-            }
-
-            if (300 <= $statusCode) {
-                throw $this->resolveResult = new RedirectionException($this->response);
-            }
+        if (500 <= $statusCode) {
+            $this->exception = new ServerException($this->response);
```

Comment:
Just realized that this code were wrong, calling this method several time with $throw=false then true does not return the rigth result. You don't need `this->exception`, we can use `this->resolveResult` to store the exception.

I think a best fix would be to remove the `$throw` parameter in this class,  and use try/catch in waiter to get the exception and silent errors

Reply:
I've removed the `$throw` argument.